### PR TITLE
Settler Fills

### PR DESCRIPTION
**Pull request by bercsenyirihi**
#139

### CODE REVIEW

File: java/src/hu/holyoil/skeleton/testcases/SettlerFillsAsteroidWithUranium.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Override
+    protected void load() {
+        u = new Uranium();
+        a = new Asteroid();
+        s = new Settler(a);
+        ps = s.GetStorage();
+
+        Logger.RegisterObject(ps,"ps: PlayerStorage");
+        Logger.RegisterObject(u,"u: Urnaium");
```
Comment (triszt4n) [inline java/src/hu/holyoil/skeleton/testcases/SettlerFillsAsteroidWithUranium.java:30]: Typo: Urnaium -> Uranium

File: java/src/hu/holyoil/skeleton/testcases/SettlerTriesToFillAsteroidWithUranium.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        ps = s.GetStorage();
+
+        Logger.RegisterObject(ps, "ps: PlayerStorage");
+        Logger.RegisterObject(u, "u: Urnaium");
+        Logger.RegisterObject(u1, "u1: Uranium");
+        Logger.RegisterObject(s, "s: Settler");
+        Logger.RegisterObject(a, "a: Asteroid");
+
+        a.AddCrewmate(s);
+        ps.SetStoredMaterial(u);
```
Comment (triszt4n) [inline java/src/hu/holyoil/skeleton/testcases/SettlerTriesToFillAsteroidWithUranium.java:37]: Maybe, we should ask the tester: Does this asteroid have a resource already? And then we don't have to create 2 separate test cases for this one issue.

File: java/src/hu/holyoil/skeleton/testcases/SettlerTriesToFillAsteroidWithUranium.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    protected void load() {
+        u = new Uranium();
+        u1 = new Uranium();
+        a = new Asteroid();
+        s = new Settler(a);
+        ps = s.GetStorage();
+
+        Logger.RegisterObject(ps, "ps: PlayerStorage");
+        Logger.RegisterObject(u, "u: Urnaium");
```
Comment (triszt4n) [inline java/src/hu/holyoil/skeleton/testcases/SettlerTriesToFillAsteroidWithUranium.java:31]: Typo: Urnaium -> Uranium

**Review by triszt4n — CHANGES_REQUESTED** [review]
Maybe merge the two test case into one.

**Review by vercibar — CHANGES_REQUESTED** [review]
Looks good. One small change is needed

File: java/src/hu/holyoil/neighbour/Asteroid.java [hunk 4]
```diff
@@ -50,6 +51,17 @@ public void ReactToMove(Asteroid from, AbstractCrewmate abstractCrewmate) {
...
 
+    /** eredeti változat a szekvenciadiagramokon: SetResource(s: Settler, resource: AbstractBaseResource)
+     *
+     */
+    public void PutResource(Settler s, AbstractBaseResource res) {
+        Logger.Log(this, "Putting down resource from settler to asteroid core.");
+        if (numOfLayersRemaining == 0 && resource == null) {
+            res.ReactToPlace(this, s);
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (triszt4n) [reply java/src/hu/holyoil/neighbour/Asteroid.java:61]: true, megoldom

**Review by triszt4n — APPROVED** [review]

**Review by vercibar — APPROVED** [review]

**Pull request merged**